### PR TITLE
Bluetooth: Convert to new logger subsystem

### PR DESCRIPTION
**Pull request by jhedberg**
Convert the Bluetooth subsystem to the new logger APIs.

Known issues:

 - Some bt_hex() uses (especially the longer chunks of data) should be replaced with `LOG_HEXDUMP()`
 - BT_* should be renamed to LOG_* but this depends on function name support (see #10644)

### CODE REVIEW

**Comment by dleach02** [comment]
There appears to be a lot more changes here outside of the logging. Is the intent to separate the other changes and have a logging only change made available at some point?

File: subsys/bluetooth/host/monitor.c [hunk 1]
```diff
@@ -241,6 +216,108 @@ extern void __stdout_hook_install(int (*fn)(int));
...
+{
+	struct monitor_log_ctx *ctx = user_data;
+	size_t i;
+
+	for (i = 0; i < length && ctx->total_len < sizeof(ctx->msg); i++) {
+		/* FIXME: There should be a way to request the log subsystem
+		 * not to add these to the end of messages.
+		 */
+		if (data[i] == '\r' || data[i] == '\n') {
+			break;
```
Comment (jhedberg) [inline subsys/bluetooth/host/monitor.c]: @nordic-krch could we get a flag for the log subsystem so that I don't need to do this kind of truncation?

**Comment by jhedberg** [comment]
I've removed the DNM tag since this now builds successfully and the monitor back end works. Strings are also handled using log_strdup() however some bt_hex calls should probably still be converted to LOG_HEXDUMP(). I'd also like to rename all BT_* calls to corresponding LOG_* ones, however that may require waiting for a resolution for #10644 since e.g. BT_DBG and BT_ERR include the function name in the output.

**Review by nashif — DISMISSED** [review]

**Comment by nashif** [comment]
looks like you missed a few samples/tests?

```
samples/bluetooth/hci_spi/prj.conf
samples/bluetooth/hci_spi/src/main.c
samples/bluetooth/hci_uart/src/main.c
samples/bluetooth/ipsp/prj.conf
samples/bluetooth/ipsp/prj_dbg.conf
samples/bluetooth/ipsp/prj_zep1656.conf
samples/bluetooth/ipsp/src/main.c
samples/boards/nrf52/mesh/onoff-app/prj.conf
tests/bluetooth/tester/src/bttester.c
tests/bluetooth/tester/src/bttester.h
tests/bluetooth/tester/src/gap.c
tests/bluetooth/tester/src/gatt.c
tests/bluetooth/tester/src/l2cap.c
tests/bluetooth/tester/src/mesh.c
```

**Comment by jhedberg** [comment]
> looks like you missed a few samples/tests?

Yep, I was mainly focusing on the stack rather than sample apps and missed these. I'll work on getting them converted.

**Comment by jhedberg** [comment]
@nashif I've updated the apps now

**Review by carlescufi — CHANGES_REQUESTED** [review]
We are still using `printk` in the built-in samples/bluetooth/*. Is this intended? it looks pretty ugly to me:

```
***** Booting Zephyr OS zephyr-v1.13.0-1191-gfbe43203b2 *****
Bluetooth initialized
Advertising successfully started
[00:00:00.007,629] <inf> bt_hci_core: HW Platform: Nordic Semiconductor (0x0002)
[00:00:00.007,659] <inf> bt_hci_core: HW Variant: nRF52x (0x0002)
[00:00:00.007,659] <inf> bt_hci_core: Firmware: Standard Bluetooth controller (0x00) Version 1.13 Build 99
[00:00:00.007,904] <wrn> bt_hci_core: bt_init: No ID address. Expecting one to come from storage.
[00:00:00.013,336] <inf> bt_hci_core: Identity: fb:d0:38:54:d1:7a (random)
[00:00:00.013,336] <inf> bt_hci_core: HCI: version 5.0 (0x09) revision 0x0000, manufacturer 0x05f1
[00:00:00.013,336] <inf> bt_hci_core: LMP: version 5.0 (0x09) subver 0xffff
```

**Review by carlescufi — APPROVED** [review]
Seems like using `printk` for sample output is the standard behavior right now in Zephyr, so no reason for me to nack this PR for this.

**Review by nashif — APPROVED** [review]

**Review by nordic-krch — APPROVED** [review]

**Pull request merged**